### ISSUE TITLE
Crash when planning for sub-queries that have grouping sets

### ISSUE THREAD

**Issue by guofengrichard** [greenplum-db/gpdb#10813]
### Greenplum version or build
6X

### Step to reproduce the behavior

```
# explain (costs off) WITH table1 AS (
     SELECT 2 AS city_id, 5 AS cnt
     UNION ALL
     SELECT 2 AS city_id, 1 AS cnt
     UNION ALL
     SELECT 3 AS city_id, 2 AS cnt
     UNION ALL
     SELECT 3 AS city_id, 7 AS cnt
),
fin AS (
SELECT
    coalesce(country_id, city_id) AS location_id,
    total
FROM (
SELECT
1 as country_id,
city_id,
sum(cnt) as total
FROM table1
GROUP BY GROUPING SETS (1,2)
) base
)
SELECT *
FROM fin
WHERE location_id = 1;
server closed the connection unexpectedly
```

**Comment by guofengrichard** [comment]
This repro query comes from upstream thread: https://www.postgresql.org/message-id/flat/16585-9d8c340d23ade8c1%40postgresql.org

This upstream thread exposes a wrong results issue of PostgreSQL, which has been fixed recently by Tom in commit 4d346def1555ea55b3adf76fc4afa3d3495ecfdd

**Comment by guofengrichard** [comment]
I believe the culprit is how `rollup_subplans` is generated in function `plan_list_rollup_plans()`. For the first rollup, the rollup_subplan is retrieved from `root->simple_rel_array[i]->subplan`, which in this case is actually an `Append` plannode. Then later in function `rebuild_append_simple_rel_and_rte()`,  all the `rollup_subplans` are treated as SubqueryScan plannode . As a result, when this function tries to change the scanrelid in SubqueryScan, it actually changes the `appendplans` to a invalid value.

The related code snippets:

```
                if (root->simple_rel_array[i] != NULL)
                {
                    rollup_subroots = lappend(rollup_subroots, root->simple_rel_array[i]->subroot);
                    rollup_subplans = lappend(rollup_subplans, root->simple_rel_array[i]->subplan);   <----- Append node
                }
```  


```
        sroot = (PlannerInfo *)lfirst(l2);
        splan = (SubqueryScan *)lfirst(l3);

        rel->subroot = sroot;
        if (splan != NULL)
        {
            splan->scan.scanrelid = i;             <------- changes Append->appendplans
            rel->subplan = splan->subplan;
        }
```
